p3/plot/_cascade.py

```diff
@@ -6,225 +6,8 @@
-
-        # Draw a line behind the boxes to represent the application
-        endpoints = [0, len(platforms) + 1]
-        ax.plot(
-            endpoints,
-            len(endpoints) * [(i + 0.5) * fac],
-            color=app_colors[app_name],
-            marker=app_markers[app_name],
-            markersize=8,
-            lw=1.5,
-            zorder=1,
-        )
-
-        # Plot a box for each platform supported by this application
-        supported_platforms = list(app_df["platform"])
-        xvalues = np.arange(1, len(supported_platforms) + 1)
-        boxes = ax.bar(
-            xvalues,
-            height=fac,
-            width=1,
-            bottom=i * fac,
-            edgecolor="black",
-            color=[plat_colors[p] for p in supported_platforms],
-            linewidth=1,
-            zorder=2,
-        )
-
-        # Label the box with its associated platform
-        for j, box in enumerate(boxes):
-            ax.text(
-                box.get_x() + box.get_width() * 0.5,
-                box.get_y() + box.get_height() * 0.5,
-                plat_labels[supported_platforms[j]],
-                ha="center",
-                va="center",
-                c="black",
-                zorder=3,
-            )
-
-
-def _pp_bars(ax, df, pp_column, colors, markers):
-    """
-    Plot a bar for the performance portability of each application,
-    using the axes provided.
-    """
-    applications = df["application"].unique()
-
-    ax.set_xticks([])
-    ax.set_ylabel("Performance Portability", rotation=-90, labelpad=14)
-    ax.set_ylim([0, 1.1])
-    ax.yaxis.set_label_position("right")
-    ax.yaxis.tick_right()
-    ax.grid(visible=True)
-
-    pp = p3.metrics.pp(df)
-
-    edgecolors = [colors[app] for app in pp["application"]]
-    ax.bar(
-        pp["application"],
-        pp[pp_column],
-        color="white",
-        edgecolor=edgecolors,
-        zorder=3,
-    )
-
-    for app_name in applications:
-        app_pp = pp[(pp["application"] == app_name)]
-        ax.plot(
-            app_name,
-            app_pp[pp_column],
-            color=colors[app_name],
-            marker=markers[app_name],
-            markersize=8,
-            lw=1.5,
-            zorder=4,
-        )
-
-
-def _get_colors(applications, kwarg):
-    """
-    Assign a color to each application based on supplied kwarg.
-    """
-    if isinstance(kwarg, str):
-        cmap = getattr(plt.cm, kwarg)
-    elif isinstance(kwarg, list):
-        cmap = matplotlib.colors.ListedColormap(kwarg)
-    elif isinstance(kwarg, matplotlib.colors.Colormap):
-        cmap = kwarg
-    else:
-        raise ValueError("Unsupported type provided for colormap.")
-
-    cmap = cmap.resampled(len(applications))
-    colors = cmap(np.linspace(0, 1, len(applications)))
-    return {app: color for app, color in zip(applications, colors)}
+from p3._utils import _require_columns
+from p3.plot.backend.matplotlib import CascadePlot as MPLCascadePlot
```

Comment:
Doing things this way means that matplotlib has to be installed even when the user is directing things to another backend.  With multiple backends, the user would have to install all the backends just to use one of them.

So we should remove this:
```suggestion
```

See below for my suggested fix.  Again, note that it should apply to the equivalent lines for navigation charts too.

Reply:
Agree. Done.